text/0808-library-quality-scoring.md

```diff
@@ -0,0 +1,633 @@
+ --verbose  Show detailed breakdown of signals
+ --help     Show this help message
+```
+
+You can run it locally on any library published to npm by providing its package name:
+
+```> cdk-construct-analyzer @cdklabs/cdk-ecs-codedeploy
+
+LIBRARY: @cdklabs/cdk-ecs-codedeploy
+VERSION: 0.0.421
+
+OVERALL SCORE: 75/100
+
+---
+
+SUBSCORES
+  Maintenance :            66/100
+  Quality     :            76/100
+  Popularity  :            83/100
+```
+
+Add `--verbose` for a detailed breakdown:
+
+```> cdk-construct-analyzer @cdklabs/cdk-ecs-codedeploy --verbose
+
+LIBRARY: @cdklabs/cdk-ecs-codedeploy
+VERSION: 0.0.421
+
+OVERALL SCORE: 75/100
+
+---
+
+SUBSCORES
+  Maintenance :            66/100
+  Quality     :            76/100
+  Popularity  :            83/100
+  
+---
+
+=== Maintenance ===                               SCORE  WEIGHT
+— Time to first response......................... ★★★☆☆    3
+— Commit Frequency .............................. ★★★☆☆    3
+— Release Frequency ............................. ★★★★☆    3
+— Provenance Verification ....................... ★★★★★    3
+— Open issues / total issues .................... ★★★★★    3
+— Median PR time-to-merge ....................... ★★☆☆☆    2
+— Number of Contributors ........................ ★★★☆☆    2
+— Most recent commit ............................ ★★★★☆    2
+
+=== Quality ===                                   SCORE  WEIGHT
+— README, API Reference and usage examples ...... ★★★★★    3
+— Tests checklist (unit/snapshot) ............... ★★★☆☆    3
+— Passing CI builds ............................. ★★★★★    3
+— Author Track Record ........................... ★★★★★    3
+— Changelog Present ............................. ★☆☆☆☆    3
+— Stable versioning ............................. ★★★★☆    2
+— License, .gitignore/.npmignore present ........ ★★★★☆    1
+— Multi-language Support ........................ ★★★☆☆    1
+
+=== Popularity ===                                SCORE  WEIGHT
+— Weekly Downloads .............................. ★★★★★    3
+— Repo stars .................................... ★★★★☆    2
+— Contributors .................................. ★★★☆☆    1
+```
+
+#### Scoring Pillars and Signals
+
+The scoring algorithm evaluates each construct on three pillars with multiple weighted signals as support:
+
+##### Maintenance
+
+Helps determine if the project is active and healthy, or abandoned. Signals include:
+
+* Time to first response: Fast issue resolution reflects active, responsive maintainers.
+* Commit Frequency: Frequent commits show steady development and maintenance.
+* Release Frequency: Regular releases signal iteration, patching, and progress.
+* Provenance Verification: Verifies package authenticity and supply chain security.
+* Open issues / total issues: A lower ratio of open issues indicates backlog health and follow-through
+normalized by repository popularity.
+* Median PR time-to-merge: Shorter merge times suggest attentiveness to community contributions.
```

Comment:
Same commit frequency metric yet again? Is this a sliding window? What's the window? All PRs or contributor PRs?